pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/PinotURIUploadIntegrationTest.java

```diff
@@ -186,6 +186,38 @@ private void verifyNRows(int currentNrows, int finalNrows)
     Assert.fail("Failed to get from " + currentNrows + " to " + finalNrows);
   }
 
+  @Test(dataProvider = "configProvider")
+  public void testSegmentValidator(String tableName, SegmentVersion version)
+      throws Exception {
+    completeTableConfiguration();
+    String serverInstanceId = "Server_localhost_" + CommonConstants.Helix.DEFAULT_SERVER_NETTY_PORT;
+
+    // Disable server instance.
+    _helixAdmin.enableInstance(getHelixClusterName(), serverInstanceId, false);
+
+    final String segment6 = "segmentToBeRefreshed_6";
+    final int nRows1 = 69;
+    File segmentTarDir = generateRandomSegment(segment6, nRows1);
+    try {
+      uploadSegmentsDirectly(segmentTarDir);
+      Assert.fail("Uploading segments should fail.");
+    } catch (Exception e) {
+      Assert.assertNotNull(e);
+      Assert.assertTrue(e.getMessage().contains("No assigned Instances for Segment"));
```

Comment:
Don't check the exception message content. We might change the exception message in the future and that should not fail the test. The fail() in try block should be enough

Reply:
Done.